src/code/flg_set.c

```diff
@@ -33,12 +28,126 @@ FlagReference D_80115DC0[53] = {
     { &gSaveContext.infTable[28], "inf_table[28]" },        { &gSaveContext.infTable[29], "inf_table[29]" },
     { &gSaveContext.eventInf[0], "event_inf[0]" },          { &gSaveContext.eventInf[1], "event_inf[1]" },
     { &gSaveContext.eventInf[2], "event_inf[2]" },          { &gSaveContext.eventInf[3], "event_inf[3]" },
-};
+} };
+
+void FlagSet_Update(GlobalContext* globalCtx) {
+    static s32 entryIdx = 0;
+    static u32 curBit = 0;
+    static s32 timer = 0;
+    static s32 bitIdx;
+
+    FlagSetEntryHolder holder;
+    GraphicsContext* gfxCtx;
+    u32 pad;
+    Gfx* gfx;
+    Gfx* polyOpa;
+    Gfx* dispRefs[5];
+    GfxPrint printer;
+    Input* input = &globalCtx->state.input[0];
+
+    holder = sFlagSetEntries;
```

Comment:
Since there is a full copy here, any chance the flag set entry array could be defined inside the function itself? (and thus no need for it to be a struct)